lib/langchain/conversation.rb

```diff
@@ -59,7 +59,19 @@ def add_examples(examples)
     def message(message)
       human_message = HumanMessage.new(message)
       @memory.append_message(human_message)
-      ai_message = llm_response(human_message)
+      ai_message = llm_response
+      @memory.append_message(ai_message)
+      ai_message
+    end
+
+    # Message the model with the function name, content and return the response.
+    # @param message [String] The content or arguments for the function call
+    # @param name [String] The name of the function to be called
```

Comment:
If I understand these comments correctly (e.g. "the function to be called") you expect that the LLM somehow invoke the function, am I right?


In [step 3 of this example](https://openai.com/blog/function-calling-and-other-api-updates) we can see a function message and it looks like
`{"role": "function", "name": "get_current_weather", "content": "{\"temperature\": "22", \"unit\": \"celsius\", \"description\": \"Sunny\"}"}`

which I understand as a function that was already invoked and the content carries its result.

Reply:
Not exactly the result, but what you want the LLM to understand. For example, if you write on the content "API error", the LLM will process and generate a message to the user. So, it depends on the case.